actors/builtin/storage_power/storage_power_actor.go

```diff
@@ -37,44 +38,49 @@ type StoragePowerActor struct{}
 // Actor methods
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StoragePowerActor) AddBalance(rt Runtime, minerAddr addr.Address) *adt.EmptyValue {
-	builtin.RT_MinerEntry_ValidateCaller_DetermineFundsLocation(rt, minerAddr, builtin.MinerEntrySpec_MinerOnly)
+func (a *StoragePowerActor) Constructor(rt Runtime) *adt.EmptyValue {
```

Comment:
I moved the ctor to the top, and pushed the state initialization into the state type.